pkg/apis/ark/v1/backup.go

```diff
@@ -56,6 +59,9 @@ type BackupSpec struct {
 
 	// Hooks represent custom behaviors that should be executed at different phases of the backup.
 	Hooks BackupHooks `json:"hooks"`
+
+	// StorageLocation is a reference to a BackupStorageLocation where the backup should be stored.
+	StorageLocation corev1api.ObjectReference `json:"storageLocation"`
```

Comment:
Could this just be a `string` for the name of the `BackupStorageLocation`?

Reply:
@ncdc is there a best practice or pattern around when to use a string/strings vs. `LocalObjectReference` vs. `ObjectReference`?

Reply:
`ObjectReference` is if you need to go across namespaces. It supports `APIVersion` and `Kind` for specifying different resources.

`LocalObjectReference` currently only has a `Name` field, so you can't specify `APIVersion` and `Kind` in the event you needed to be able to reference different resources.

As to `string` vs `LocalObjectReference`, I don't know that there's any good guidance. If `LocalObjectReference` supported `APIVersion` and `Kind`, I'd say use `string` if you don't need to vary the resource, and `LocalObjectReference` if you do. Given that this isn't a supported use case, and we only care about a single kind of resource (`BackupStorageLocation`), my recommendation would be to use a `string` here. It also makes the json/yaml easier to write.

WDYT - do you agree?

Reply:
That makes sense - a `string` seems fine to me.

Reply:
yeah, that works for me. I was thinking this was a `LocalObjectRef` when i reviewed it, and i'm fine just using a string in place.